images/macos/provision/core/configure-toolset.ps1

```diff
@@ -0,0 +1,47 @@
+################################################################################
+##  File:  Configure-Toolset.ps1
+##  Team:  CI-Build
+##  Desc:  Configure toolset
+################################################################################
+
+Import-Module "~/image-generation/helpers/Common.Helpers.psm1"
+
+function Get-ToolsetToolFullPath
+{
+    param
+    (
+        [Parameter(Mandatory)] [string] $ToolName,
+        [Parameter(Mandatory)] [string] $ToolVersion,
+        [Parameter(Mandatory)] [string] $ToolArchitecture
+    )
+
+    $toolPath = Join-Path -Path $env:AGENT_TOOLSDIRECTORY -ChildPath $toolName
+    $toolPathVersion = Join-Path -Path $toolPath -ChildPath $toolVersion
+    $foundVersion = Get-Item $toolPathVersion | Sort-Object -Property {[version]$_.name} -Descending | Select-Object -First 1
+    $installationDir = Join-Path -Path $foundVersion -ChildPath $toolArchitecture
+    return $installationDir
+}
+
+$toolcache = Get-ToolsetValue "toolcache"
+
+foreach ($tool in $toolcache)
+{
+    $toolName = $tool.name
+    $toolArch = $tool.arch
```

Comment:
Can we have multiple arch?

Reply:
Discussed offline, decided to leave it as it is